metadata-ingestion/tests/unit/s3/test_s3_config_profiling.py

```diff
@@ -0,0 +1,243 @@
+"""Unit tests for S3 config profiling validation."""
+
+import pytest
+
+# Check if profiling dependencies are available
+try:
+    import pydeequ  # noqa: F401
+    import pyspark  # noqa: F401
+
+    _PROFILING_ENABLED = True
+except ImportError:
+    _PROFILING_ENABLED = False
+
+from datahub.ingestion.source.s3.config import DataLakeSourceConfig
+
+
+class TestS3ConfigProfilingValidation:
+    """Tests for S3 config profiling dependency validation."""
+
+    def test_config_without_profiling(self):
+        """Test that S3 config can be created without profiling enabled."""
+        config_dict: dict = {
+            "path_specs": [
+                {
+                    "include": "s3://test-bucket/data/*.parquet",
+                }
+            ],
+            "profiling": {"enabled": False},
+        }
+
+        config = DataLakeSourceConfig.parse_obj(config_dict)
+
+        assert config is not None
+        assert config.platform == "s3"
+        assert config.profiling.enabled is False
+
+    def test_config_profiling_disabled_by_default(self):
+        """Test that profiling is disabled by default."""
+        config_dict: dict = {
+            "path_specs": [
+                {
+                    "include": "s3://test-bucket/data/*.parquet",
+                }
+            ],
+        }
+
+        config = DataLakeSourceConfig.parse_obj(config_dict)
+
+        assert config is not None
+        assert config.profiling.enabled is False
+
+    def test_config_with_profiling_when_pyspark_available(self):
+        """Test that config accepts profiling when PySpark is available."""
+        if not _PROFILING_ENABLED:
+            pytest.skip("PySpark not available, skipping test")
+
+        config_dict: dict = {
+            "path_specs": [
+                {
+                    "include": "s3://test-bucket/data/*.parquet",
+                }
+            ],
+            "profiling": {"enabled": True},
+        }
+
+        config = DataLakeSourceConfig.parse_obj(config_dict)
+
+        assert config is not None
+        assert config.profiling.enabled is True
+
+    def test_config_with_profiling_accepts_without_pyspark(self):
+        """Test that config accepts profiling even without PySpark (backward compatibility).
+
+        Note: In the default s3/gcs/abs installation, PySpark is included.
+        When using s3-slim/gcs-slim/abs-slim, profiling will be disabled at runtime
```

Comment:
Please update comments to not include gcs-slim/abs-slim references. Those do not exist in this PR.